operator/src/main/resources/scripts/introspectDomain.py

```diff
@@ -1664,6 +1662,50 @@ def getSSLOrNone(server):
     ret = None
   return ret
 
+# Derive the default value for SecureMode of a domain
+def isSecureModeEnabledForDomain(domain):
+  secureModeEnabled = false
+  if domain.getSecurityConfiguration().getSecureMode() != None:
+    secureModeEnabled = domain.getSecurityConfiguration().getSecureMode().isSecureModeEnabled()
+  else:
+    secureModeEnabled = domain.isProductionModeEnabled() and not LegalHelper.versionEarlierThan(domain.getDomainVersion(), "14.1.2.0")
+  return secureModeEnabled
+
+def isAdministrationPortEnabledForDomain(domain):
+  administrationPortEnabled = false
+  #"if domain.isSet('AdministrationPortEnabled'):" does not work in off-line WLST!
+  # Go to the domain root
+  cd('/')
+  if isSet('AdministrationPortEnabled'):
+    administrationPortEnabled = domain.isAdministrationPortEnabled()
+  else:
+    # AdministrationPortEnabled is not explicitly set so going with the default
+    # Starting with 14.1.2.0, the domain's AdministrationPortEnabled default is derived from the domain's SecureMode
+    administrationPortEnabled = isSecureModeEnabledForDomain(domain)
+  return administrationPortEnabled
+
+def isAdministrationPortEnabledForServer(server, domain, isServerTemplate=False):
+  administrationPortEnabled = false
+  #"if server.isSet('AdministrationPortEnabled'):" does not work in off-line WLST!
+  cd('/')
+  if isServerTemplate:
+    cd('ServerTemplate')
+  else:
+    cd('Server')
+  cd(server.getName())
+  if isSet('AdministrationPortEnabled'):
+    administrationPortEnabled = server.isAdministrationPortEnabled()
+  else:
+    administrationPortEnabled = isAdministrationPortEnabledForDomain(domain)
+  return administrationPortEnabled
+
```

Comment:
I've found three additional attributes that this script checks, that have different default and secureValue fields, and that aren't included in this update:

- NetworkAccessPoint.getProtocol: t3 --> t3s (search for "getProtocol()")
- Server[Template].isListenPortEnabled: true --> false (search for "isListenPortEnabled()")
- SSL.isEnabled: false --> true (search for "isEnabled()")

I assume the script needs to be updated for them too?

Reply:
(The current pull looks excellent to me otherwise BTW.)

Reply:
Good catch.  Let me add some updates for those as well. Thanks!

Reply:
Looks good!